core/src/main/scala/monocle/std/Cofree.scala

```diff
@@ -0,0 +1,74 @@
+package monocle.std
+
+import monocle.function._
+import monocle.{Lens, PIso, Iso, Optional, PTraversal, Traversal}
+
+import scalaz.Cofree._
+import scalaz.{Cofree, Free, Applicative, Traverse, OneAnd, Tree}
+
+object cofree extends CofreeOptics
+
+trait CofreeOptics {
+
+  /** Polymorphic traversal over all the head values of a cofree structure
+    * assuming the cofree's branching functor is traversable */
+  def pHeads[S[_] : Traverse, A, B]: PTraversal[Cofree[S, A], Cofree[S, B], A, B] =
```

Comment:
I believe `pHeads` go through all elements of `Cofree[S, A]`. If that's correct, we could instead create an instance of `Each` for `Cofree`. It is only monomorphic but it is relatively easy to create a `Traversal` or `PTraversal` from `Traverse` using `fromTraverse` https://github.com/julien-truffaut/Monocle/blob/master/core/src/main/scala/monocle/Traversal.scala#L189